tests/unit/test_charm.py

```diff
@@ -58,12 +46,12 @@ def test__on_upgrade_charm(harness: ops.testing.Harness, monkeypatch: pytest.Mon
     monkeypatch.setattr(service.JenkinsAgentService, "restart", MagicMock())
     harness.begin()
 
-    jenkins_charm: JenkinsAgentCharm = harness.charm
+    charm: JenkinsAgentCharm = harness.charm
     upgrade_charm_event = MagicMock(spec=ops.UpgradeCharmEvent)
-    jenkins_charm._on_upgrade_charm(upgrade_charm_event)
+    charm._on_upgrade_charm(upgrade_charm_event)
```

Comment:
Can you test by triggering this events instead?

Reply:
yes, I updated the code to trigger the event!